From 30914db95d1c15fdb0b8f948a3631b1464fecee2 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 24 Jul 2017 13:29:58 +0800
Subject: [PATCH 2197/5242] MLK-16075-2 gpu: imx: dpu: prv: Cleanup
 definitions for YUV conversion mode bits

commit  700b6e1e3e508acdeb053305e3e09ab908ed50bc from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch cleans up definitions for the YUV conversion mode register field.
Two macros are introduced for users to program the field easily.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-prv.h |   12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/imx/dpu/dpu-prv.h b/drivers/gpu/imx/dpu/dpu-prv.h
index caf95d1..4dd87c3 100644
--- a/drivers/gpu/imx/dpu/dpu-prv.h
+++ b/drivers/gpu/imx/dpu/dpu-prv.h
@@ -74,11 +74,13 @@ enum linemode {
 #define RGBALPHATRANSENABLE		BIT(15)
 #define PREMULCONSTRGB			BIT(16)
 typedef enum {
-	YUVCONVERSION__OFF,
-	YUVCONVERSION__ITU601,
-	YUVCONVERSION__ITU601_FR,
-	YUVCONVERSION__ITU709,
-} yuvconversion_t;
+	YUVCONVERSIONMODE__OFF,
+	YUVCONVERSIONMODE__ITU601,
+	YUVCONVERSIONMODE__ITU601_FR,
+	YUVCONVERSIONMODE__ITU709,
+} yuvconversionmode_t;
+#define YUVCONVERSIONMODE_MASK		0x60000
+#define YUVCONVERSIONMODE(m)		(((m) & 0x3) << 17)
 #define GAMMAREMOVEENABLE		BIT(20)
 #define CLIPWINDOWENABLE		BIT(30)
 #define SOURCEBUFFERENABLE		BIT(31)
-- 
1.7.9.5

